Update Redis Dockerfile to fix wrong version compiled

aran112000 opened the pull request:
After discovering the Redis extension was failing on PHP startup for 8.1 (https://github.com/brefphp/extra-php-extensions/issues/281), I've now tracked down the extension was being compiled against PHP 8.1 **BETA** 3, rather than RC4. Removing the version on the Docker image should help to keep it in sync going forward with the PHP versions of future Bref releases.

aran112000 commented:
Once this change has been reviewed/accepted, I'll do the same for the Memcached one too which is also failing currently

aran112000 commented:
@deleugpn You're far more knowledgeable than me when trying to get everything compiling nicely. I believe the tests are being run against old PHP versions too, which is why they were passing before (when they were actually failing when deployed to AWS), now the Lambda layer versions match, the tests are failing. What's considered the *right* way to handle this, please?

aran112000 commented:
Morning @Nyholm; any chance you'd have a chance this week to review/merge this change so we can carry on with our testing, please? Would get great to get a release with this and the Memcached changes from last week tagged so the layers go up :)

Nyholm commented:
Hey. 
This looks good. 

Could you follow up this PR with the same change on all the layers?

aran112000 commented:
> Could you follow up this PR with the same change on all the layers?

Sure thing, out of interest is there are reason versions are explicitly specified, rather than just using the latest version of the base images?

Nyholm commented:
It is generally a bad practice to use `latest` since we dont know if latest is `master`, "the highest version" or "the next version". It is just the latest build. That is also no support for semver. 

We could do `:1`, but at the time I dont think Bref tagged major version only. Do they do that now?

So the short answer is: `latest` is bad, hence we fixed the version.

aran112000 commented:
> So the short answer is: `latest` is bad, hence we fixed the version.

Makes total sense, I've pushed out a new PR updating the version on all other layers now, I did catch two that were using latest already which I've updated to use 1.3.3 instead based on this.

Thanks for your help and the quick release
